File: shopy-seller/lib/models/store/store_address.dart
class StoreAddress {
  final String id;
  final String label;
  final String picName;
  final String phoneNumber;
  final String fullAddress;
  final String city;
  final String province;
  final String postalCode;
  final bool isDefault;

  const StoreAddress({
    required this.id,
    required this.label,
    required this.picName,
    required this.phoneNumber,
    required this.fullAddress,
    required this.city,
    required this.province,
    required this.postalCode,
    required this.isDefault,
  });

  StoreAddress copyWith({bool? isDefault}) {
    return StoreAddress(
      id: id,
      label: label,
      picName: picName,
      phoneNumber: phoneNumber,
      fullAddress: fullAddress,
      city: city,
      province: province,
      postalCode: postalCode,
      isDefault: isDefault ?? this.isDefault,
    );
  }

  factory StoreAddress.fromJson(Map<String, dynamic> json) {
    return StoreAddress(
      id: json['id'] as String,
      label: json['label'] as String,
      picName: json['picName'] as String,
      phoneNumber: json['phoneNumber'] as String,
      fullAddress: json['fullAddress'] as String,
      city: json['city'] as String,
      province: json['province'] as String,
      postalCode: json['postalCode'] as String,
      isDefault: json['isDefault'] as bool,
    );
  }
}

File: shopy-api/Models/StoreAddress.cs
namespace shopy_api.Models;

public class StoreAddress : ISoftDeletable
{
    public Guid Id { get; set; }

    public Guid StoreId { get; set; }
    public Store Store { get; set; } = null!;

    public string Label { get; set; } = string.Empty;
    public string PicName { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string FullAddress { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Province { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public bool IsDefault { get; set; }

    public bool IsDeleted { get; set; }
}
